apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.js

```diff
@@ -121,6 +122,39 @@ async function processDeleteFile(req, res) {
   await deleteFile(file.id, res.locals.authn_user.user_id);
 }
 
+/**
+ * This is used to conditionally display/permit a shortcut to delete the
+ * assessment instance. Usually, the only way to delete an assessment instance
+ * is from the "Students" tab of an assessment. However, when a staff member is
+ * iterating on or testing an assessment, it can be tedious to constantly go
+ * back to that page to delete the instance in order to recreate it.
+ *
+ * The shortcut is a "Delete assessment instance" button on the assessment
+ * instance page. It's only displayed if the user has the necessary permissions:
+ *
+ * - The user must have "view" permission for the course instance.
+ * - The user must own the assessment instance, either directly or via group membership.
+ *
+ * Note that we do not check for course permissions. This addressed the case
+ * where some user was an enrolled student in course instance X and was later
+ * added as course staff to course instance Y. In this case, the user should
+ * not be able to delete their old assessment instances in course instance X.
+ *
+ * We also do not require the user to have "edit" permission for the course
+ * instance. A TA could be assigned to test an upcoming assessment, but may
+ * only have "view" permission for the course instance. In this specific case,
+ * it'd still be helpful for the TA to be able to delete and recreate their
+ * assessment instances to better multiple variants of the assessment.
+ *
+ * @returns {boolean} Whether or not the user should be allowed to delete the assessment instance.
+ */
+function canDeleteAssessmentInstance(resLocals) {
+  return (
+    resLocals.authz_data.has_course_instance_permission_view &&
```

Comment:
Do we want to check `authz_data.authn_has_course_instance_permission_view` instead? This would allow the menu to appear for both "student view" and "student view without access restrictions".

Reply:
I believe so (and changed it in 408d7fe1c). If I'm the instructor, I want to see the menu in any student view. Sometimes when I'm demoing assignments for students, I use 'student view' so they see the screens I see and I still want that delete button when I'm done.

Reply:
Thanks, I updated the comment in aab13fc to document that using `authn_*` was a deliberate choice.